include/systems/generic_projector.h

```diff
@@ -1179,17 +1345,37 @@ void GenericProjector<FFunctor, GFunctor, FValue, ProjectionAction>::SubFunctor:
     }
 }
 
+template <typename FFunctor, typename GFunctor, typename FValue, typename ProjectionAction>
+template <typename InsertInput,
+          typename std::enable_if<
+              !std::is_same<typename ProjectionAction::InsertInput, InsertInput>::value,
+              int>::type>
+void
+GenericProjector<FFunctor, GFunctor, FValue, ProjectionAction>::SubFunctor::insert_ids(
+    const std::vector<dof_id_type> &,
+    const std::vector<InsertInput> &,
+    processor_id_type)
+{
+  libmesh_error_msg("ID insertion should only occur when the provided input matches that "
+                    "expected by the ProjectionAction");
```

Comment:
This is weird.  Couldn't you have just left off this specialization, and then any mismatch would have been a compile-time error rather than a run-time error?

Reply:
If I don't have this, then I get compile errors right now in  `ProjectVertices`, `ProjectEdges`, `ProjectSides`, and `ProjectInteriors` when I instantiate a vector projection because I thought we were open to using the same calling sites for both scalar and vector field variables.

Reply:
Oh, I get it.  You're choosing which to call with an if test, so we still get an instantiation of the incompatible option, but that's okay because its an instantiation we'll never hit.